formula-android/src/main/java/com/instacart/formula/android/FeatureFactory.kt

```diff
@@ -35,10 +35,22 @@ package com.instacart.formula.android
  * @param Key a type of fragment key that is used to identify this feature.
  *
  */
-interface FeatureFactory<in Dependencies, in Key : FragmentKey> {
+abstract class FeatureFactory<in Dependencies, in Key : FragmentKey> {
+
+    inner class Params(
+        val dependencies: @UnsafeVariance Dependencies,
+        val key: @UnsafeVariance Key,
+    )
+
+    /**
+     * Initializes the [Feature] using [Params] provided.
+     */
+    abstract fun Params.initialize(): Feature
 
     /**
      * Initializes state observable and a view factory for a specific [key].
      */
-    fun initialize(dependencies: Dependencies, key: Key): Feature
+    fun initialize(dependencies: Dependencies, key: Key): Feature {
+        return Params(dependencies, key).initialize()
+    }
```

Comment:
Will this become `private` in a follow up?

Reply:
Probably not, this function provides a better API to initialize the `Feature`. It's a final function that delegates to a better internal API for implementation. 
- for implentee use `Params.initialize` 
- for the consumer use `initialize(dependencies, key, other params)`